src/test/java/parser/ParserTest.java

```diff
@@ -91,4 +88,57 @@ public void testParseGraphMissingKeywords() {
         String demo = "{graph {A B C} {{A bar B} {B foo C}}}";
         Parser.parseGraph(demo);
     }
+
+    @Test
+    public void testEdgeParseBidirectional() {
+        // needs to be casted to Edge
+        assertThat((Edge) Parser.parseEdge("{A to B}")).isEqualTo(A_TO_B);
+
+        assertThat(Parser.parseEdges("{{A to B} {B to C} {C to A}}")).isEqualTo(EDGES);
+
+        // new bi-directional test.
+        List<Edge> e1 = ImmutableList.of(Edge.of(A, B, null), Edge.of(B, A, null), Edge.of(C, A, null));
+        assertThat(Parser.parseEdges("{{A <-> B} {C to A}}")).isEqualTo(e1);
+    }
+    @Test
+    public void testEdgeParseDuplicate() {
+
+        assertThat(Parser.parseEdges("{{A to B} {A to B} {B to C} {B to C} {C to A}}")).isEqualTo(EDGES);
+
+    }
+    @Test
+    public void testEdgeParseDuplicateB() {
+        List<Edge> e1 = ImmutableList.of(Edge.of(A, B, null), Edge.of(B, A, null), Edge.of(C, A, null));
+        assertThat(Parser.parseEdges("{{A <-> B} {A to B} {C to A}}")).isEqualTo(e1);
+        assertThat(Parser.parseEdges("{{A to B} {A <-> B} {C to A}}")).isEqualTo(e1);
+        assertThat(Parser.parseEdges("{{A <-> B} {A <-> B} {B <-> A} {C to A}}")).isEqualTo(e1);
+    }
+
+//    @Test
```

Comment:
should delete this.

Reply:
you mean this test case or the commented test that comes after at line 117

Reply:
the commented one.